File: training-data/nim/coin_change.nim
proc minCoins(coins: seq[int], amount: int): int =
  var dp = newSeq[int](amount + 1)
  for i in 1 .. amount:
    dp[i] = -1
  for n in 1 .. amount:
    for c in coins:
      if c <= n and dp[n - c] != -1:
        if dp[n] == -1 or dp[n - c] + 1 < dp[n]:
          dp[n] = dp[n - c] + 1
  result = dp[amount]

echo minCoins(@[1, 2, 5], 11)

File: training-data/prolog/coin_change.pl
min_coins(_, 0, 0) :- !.
min_coins(Coins, Amount, Result) :-
    Amount > 0,
    findall(Count,
        (member(C, Coins), C =< Amount, Amount1 is Amount - C,
         min_coins(Coins, Amount1, SubCount), Count is SubCount + 1),
        Counts),
    Counts \= [],
    min_list(Counts, Result).

:- min_coins([1,2,5], 11, Result), writeln(Result).
